commands/doit.go

```diff
@@ -125,35 +106,24 @@ func initConfig() {
 	viper.SetDefault("context", "default")
 }
 
-func findConfig() (string, error) {
-	if cfgFile != "" {
-		return cfgFile, nil
-	}
-
-	if homeDir() != "" {
-		legacyConfigPath := filepath.Join(homeDir(), ".doctlcfg")
-		if _, err := os.Stat(legacyConfigPath); err == nil {
-			msg := fmt.Sprintf("Configuration detected at %q. Please move .doctlcfg to %s",
-				legacyConfigPath, configPath())
-			warn(msg)
-		}
+func currentUser() *user.User {
+	var err error
+	if me != nil {
+		return me
 	}
+	me, err = user.Current()
+	checkErr(err)
 
-	if os.Getenv("XDG_CONFIG_HOME") != "" {
-		legacyXDGPath := filepath.Join(os.Getenv("XDG_CONFIG_HOME"), "config.yaml")
-		if _, err := os.Stat(legacyXDGPath); err == nil {
-			msg := fmt.Sprintf("Configuration detected at %q. Please move config.yaml to %s",
-				legacyXDGPath, configPath())
-			warn(msg)
-		}
-	}
+	return me
+}
 
-	ch := configHome()
-	if err := os.MkdirAll(ch, 0755); err != nil {
-		return "", err
-	}
+// in case we ever want to change this, or let folks configure it...
+func configHome() string {
+	return filepath.Join(currentUser().HomeDir, ".config", "doctl")
```

Comment:
This seems to always use `$HOME/.config` and not respect `$XDG_CONFIG_HOME`, is that really what we want? Historically, we've tried to support the XDG specification: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Reply:
Looks like the `os` package now supports the XDG specification including `os.UserConfigDir()`, but it was only added in the recent 1.13 release: https://golang.org/pkg/os/#UserConfigDir